File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    let ship = people.length
    
    people.sort((a, b) => a - b)
    
    let first = 0
    let last = ship - 1
    
    while (first < last) {
        if (people[first] + people[last] <= limit) {
            ship -= 1
            first += 1
        }
        last -= 1
    }
    
    return ship;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    ship = len(people)
    
    people.sort()
    
    first = 0
    last = len(people) - 1
    
    while first < last:
        if people[first] + people[last] <= limit:
            ship -= 1
            first += 1
        last -= 1
        
    return ship